drivers/sensor/apds9960/apds9960.c

```diff
@@ -0,0 +1,572 @@
+/* sensor_apds9960.c - driver for APDS9960 ALS/RGB/gesture/proximity sensor */
+
+/*
+
+  Copyright (c) 2017 Intel Corporation
+
+  SPDX-License-Identifier: Apache-2.0
+
+*/
+
+#include "apds9960.h"
+
+uint8_t apds9960_getmode(struct device *dev)
+{
+    uint8_t enable_value;
+    struct apds9960_data *data = dev->driver_data;
+
+    /* Read current ENABLE register */
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS, 0x80 ,&enable_value) < 0){
+                SYS_LOG_DBG(" Failed");
+		return 0xFF;
+        }
+
+    return enable_value;
+}
+
+bool apds9960_setmode(struct device *dev, uint8_t mode, uint8_t enable)
+{
```

Comment:
This function seems unnecessarily complicated. Why not just use i2c_reg_update_byte instead?